### DISCUSSION TITLE
how to add timeout into postman-to-k6 script?

### DISCUSSION THREAD

**Discussion by BrendaAspen** [apideck-libraries/postman-to-k6#60] [Q&A] [answered]
I have an API, the response time is more than 60s (60s is the default timeout setting in k6).
In this default setting, response timeout issues always pop up.
__WARN[0063] Request Failed                                error="Post \"http://10.148.82.149/api/training/Agent\": request timeout"__

I want to know if we have any options in postman-to-k6 converting?
Or, any idea on how to resolve this problem?

Thanks in advance.

**Comment by thim81** [comment, marked as answer]
The time-out is a [K6 parameter,](https://k6.io/docs/javascript-api/k6-http/params/) so you should not set that in your Postman collection.

postman-to-k6 supports the [passing of K6 params](https://github.com/apideck-libraries/postman-to-k6#k6-param-options-file)

So my advice would be to create a `k6-params.json`:

```json
{
  "timeout:": "120s",
  ... other K6 params if needed
}
```
This will set the K6 timeout to 120 seconds, instead of the default 60 seconds.

and then include the K6 params file in the conversion

`postman-to-k6 collection.json --k6-params k6-params.json -o k6-script.js`

**Reply by BrendaAspen** [reply]
Let me try, thanks for your help me again @thim81

**Comment by Fuentealba** [comment]
Hi, 

I'm doing what thim81 suggested, I get this error:

`WARN[0048] There were unknown fields in the options exported in the script  error="json: unknown field \"timeout:\""`

is there a way to add a different timeout?

Regards.

Patricio Fuentealba

**Reply by BrendaAspen** [reply]
Hi @Fuentealba , as some reason, I didn't use k6-params.json from @thim81 suggestion.
I added:
 params = {
        **timeout: "60m",**
        headers: {
            Accept: "application/json, text/plain, */*",
            "User-Agent":
                "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.75 Safari/537.36",
            "Content-Type": "application/json"
        }
    };

res = http.get(url_base + `/api/public/v1/permissions`, **params**);

With params - timeout setting for each http request, it resolved my question.

It means for the script, the total timeout is 60m for your urls.

**Comment by Fuentealba** [comment]
Yes, I found out that adding timeout in options works as well no matter the error that appears on execution.

Thank you.
